front/app/components/FormBuilder/components/FormFields/FormField/index.tsx

```diff
@@ -84,10 +84,8 @@ export const FormField = ({
 
   const hasErrors = !!errors.customFields?.[index];
 
-  const showLogicOnRow =
-    // TODO: Fix this the next time the file is edited.
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    field.input_type !== 'page' ? field.logic?.rules : field.logic;
+  // NOTE: We always show the default page logic on a page field
+  const showLogicOnRow = field.input_type !== 'page' ? field.logic.rules : true;
```

Comment:
Minor: We could make it a boolean
```suggestion
  const showLogicOnRow = field.input_type !== 'page' ? (field.logic.rules?.length ?? 0) > 0 : true;
```